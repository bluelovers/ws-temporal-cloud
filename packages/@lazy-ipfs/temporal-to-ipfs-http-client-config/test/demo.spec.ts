import { assertTemporalConfig } from '../index';

test(`assertTemporalConfig`, () =>
{

	// @ts-ignore
	expect(() => assertTemporalConfig()).toThrowErrorMatchingSnapshot();

	expect(() => assertTemporalConfig({})).toThrowErrorMatchingSnapshot();

	expect(() => assertTemporalConfig({
		ipfsEndpoint: '',
	})).toThrowErrorMatchingSnapshot();

	expect(() => assertTemporalConfig({
		ipfsEndpoint: 'xxxxxxx',
		token: '',
	})).toThrowErrorMatchingSnapshot();

	expect(() => assertTemporalConfig({
		ipfsEndpoint: 'xxxxxxx',
		token: 'yyyyyyyy',
		ipfsEndpointPort: '' as any,
	})).toThrowErrorMatchingSnapshot();

	expect(() => assertTemporalConfig({
		ipfsEndpoint: 'xxxxxxx',
		token: 'yyyyyyyy',
		ipfsEndpointPort: 'kkk' as any,
	})).toThrowErrorMatchingSnapshot();

	expect(() => assertTemporalConfig({
		ipfsEndpoint: 'xxxxxxx',
		token: 'yyyyyyyy',
		ipfsEndpointPort: 777,
	})).not.toThrow();

	expect(() => assertTemporalConfig({
		ipfsEndpoint: 'xxxxxxx',
		token: 'yyyyyyyy',
		ipfsEndpointPort: null,
	})).not.toThrow();

	expect(() => assertTemporalConfig({
		ipfsEndpoint: 'xxxxxxx',
		token: 'yyyyyyyy',
		ipfsEndpointPort: void 0,
	})).not.toThrow();

});
