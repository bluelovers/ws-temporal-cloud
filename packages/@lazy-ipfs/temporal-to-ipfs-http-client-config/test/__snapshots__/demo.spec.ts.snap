// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`assertTemporalConfig 1`] = `"temporal.ipfsEndpoint is required"`;

exports[`assertTemporalConfig 2`] = `"temporal.ipfsEndpoint is required"`;

exports[`assertTemporalConfig 3`] = `"temporal.ipfsEndpoint is required"`;

exports[`assertTemporalConfig 4`] = `"temporal.token is required"`;

exports[`assertTemporalConfig 5`] = `"temporal.ipfsEndpointPort should is port, but got ''"`;

exports[`assertTemporalConfig 6`] = `"temporal.ipfsEndpointPort should is port, but got 'kkk'"`;
